core/services/job/orm.go

```diff
@@ -214,14 +214,23 @@ func (o *orm) unclaimJob(ctx context.Context, id int32) error {
 }
 
 func (o *orm) RecordError(ctx context.Context, jobID int32, description string) {
+	// Noop if the job has been deleted.
+	err := o.db.First(&models.JobSpecV2{}, "id = ?", jobID).Error
```

Comment:
Could we get a test for this?

Reply:
I tested it via https://github.com/smartcontractkit/chainlink/pull/3627/files#diff-672e5791803c3412487b15fb5fdbb1fb87791273f2d08efb0d8c7771161f17b1R352, didn't seem worthy of an automated test but I can add it if you think we should

Reply:
Hmm, seems like you'd want to check the error record exists/doesn't after that line, otherwise it just seems to be fire and forget.

Reply:
I just looked at the log output and saw no logs after that last RecordError - you're right though super easy to a check I'll do that